ui/shared/status/core/StatusBadge.qml

```diff
@@ -0,0 +1,40 @@
+import QtQuick 2.13
+import "../../../imports"
+
+Rectangle {
+    id: statusBadge
+
+    property int value
+
+    implicitHeight: statusBadge.value > 0 ? 18 + statusBadge.border.width : 10 + statusBadge.border.width
+    implicitWidth: {
+        if (statusBadge.value > 99) {
+            return 28 + statusBadge.border.width
+        }
+        if (statusBadge.value > 9) {
+            return 26 + statusBadge.border.width
+        }
+        if (statusBadge.value > 0) {
+            return 18 + statusBadge.border.width
+        }
+        return 10 + statusBadge.border.width
+    }
```

Comment:
It's important to add `border.width` to the expected `implicitWidth`, because it otherwise includes a possible supplied `border.width` in the overall `width`, making the badge smaller than it should be.

@borismelnik lemme know if there's a different way to fix this.

Reply:
Maybe in this case we should use `width` instead of `implicitWith`?

Reply:
Yeah i think `width` is right, bc it not natural size its dynamic width. But if someone must set static width like `200` this logic will be ignored for renderer, but exists.

Reply:
Good question. As per design, we know for sure how big those badges will be at any given time. If we go with `value.width` I could imagine we might end up with different sizes when e.g. `font.weight` is configured.

Reply:
So turns out this is behaves the same with `width/height` as well. So will roll back to `implicitWidth`. Also tried going with `value.width + some padding` but it seems the `width` of `Text` elements isn't honored here..